packages/server/graphql/private/mutations/processRecurrence.ts

```diff
@@ -0,0 +1,118 @@
+import ms from 'ms'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import {RRule} from 'rrule'
+import getRethink from '../../../database/rethinkDriver'
+import MeetingTeamPrompt from '../../../database/types/MeetingTeamPrompt'
+import {getActiveMeetingSeries} from '../../../postgres/queries/getActiveMeetingSeries'
+import {MeetingSeries} from '../../../postgres/queries/getMeetingSeriesByIds'
+import {analytics} from '../../../utils/analytics/analytics'
+import publish from '../../../utils/publish'
+import standardError from '../../../utils/standardError'
+import createTeamPrompt from '../../mutations/helpers/createTeamPrompt'
+import endTeamPrompt from '../../mutations/helpers/endTeamPrompt'
+import isStartMeetingLocked from '../../mutations/helpers/isStartMeetingLocked'
+import {IntegrationNotifier} from '../../mutations/helpers/notifications/IntegrationNotifier'
+import {MutationResolvers} from '../resolverTypes'
+
+const startRecurringTeamPrompt = async (
+  meetingSeries: MeetingSeries,
+  startTime: Date,
+  dataLoader,
+  r,
+  subOptions
+) => {
+  const {teamId, facilitatorId} = meetingSeries
+
+  // AUTH
+  const unpaidError = await isStartMeetingLocked(teamId, dataLoader)
+  if (unpaidError) return standardError(new Error(unpaidError), {userId: facilitatorId})
+
+  const meeting = await createTeamPrompt(teamId, facilitatorId, r, dataLoader)
+
+  const formattedDate = new Date().toLocaleDateString('en-US', {
+    month: 'short',
+    day: 'numeric'
+  })
+
+  meeting.name = `${meetingSeries.title} - ${formattedDate}`
+  meeting.scheduledEndTime = new Date(startTime.getTime() + ms(`${meetingSeries.duration}m`))
+  meeting.meetingSeriesId = meetingSeries.id
+
+  await r.table('NewMeeting').insert(meeting).run()
+
+  IntegrationNotifier.startMeeting(dataLoader, meeting.id, teamId)
+  analytics.meetingStarted(facilitatorId, meeting, undefined, true)
+  const data = {teamId, meetingId: meeting.id}
+  publish(SubscriptionChannel.TEAM, teamId, 'StartTeamPromptSuccess', data, subOptions)
+}
+
+const processRecurrence: MutationResolvers['processRecurrence'] = async (
+  _source,
+  {},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const r = await getRethink()
+  const now = new Date()
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+
+  // VALIDATION
+
+  // RESOLUTION
+  // Find any meetings with a scheduledEndTime before now, and close them
+  const teamPromptMeetingsToEnd = (await r
+    .table('NewMeeting')
+    .filter({meetingType: 'teamPrompt'})
```

Comment:
Somewhat radical suggestion - we land this as is, then come back to fixing this as a very-near-term follow-up (like, within a week).  Reasoning:
* Recurrence is still behind its own feature flag
* This mutation isn't yet being run in prod unless a superuser executes it manually
* 6 seconds isn't _too_ bad for a non-interactive query
* The actual fix will be pretty self-contained (add the index via migration, then change this query to use the migration).
* Getting this landed will enable better end-to-end testing in dev.

LMK what you think - I'm usually pretty aggressive about minimizing scope creep once something's in review, but if you think this is important enough to take care of now, I'll do it as part of this PR

Reply:
On second thought, I just went ahead and figured out the index

Reply:
usually, anything behind a feature flag i'm totally OK with. The exception is where a feature could impact the usability for features not under the feature flag. RethinkDB **_SHOULD_** be able to handle a long-running query while also processing a bunch of updates, they even say they do! https://rethinkdb.com/docs/architecture.

In practice, we've crashed the DB doing this, where queries to single tables hang :sweat_smile:. This is the top reason why I wanna get off of RethinkDB, it's just not stable at longrunning tasks. In the future & in PG, this is totally safe

Reply:
_Oof_, that's not good. Does rethink have a timeout-like thing, a la Postgres's `statement_timeout`? If a long-running query like this can DOS the DB, we should probably raise an error anytime the duration of a query crosses some threshold.

Reply:
I don't think so... at least i haven't found it! the last couple times it was crashed with an ad-hoc query going via the explorer. It's hard to see which query causes the crash. there's a jobs table, but when the whole thing freezes, so does that table.